modules/imap/output_modules.php

```diff
@@ -101,17 +102,32 @@ protected function output() {
             }
             if (isset($struct['subtype']) && strtolower($struct['subtype']) == 'html') {
                 $allowed = $this->get('header_allow_images');
-                $images = $this->get('imap_allow_images', false);
-                if ($allowed && stripos($this->get('msg_text'), 'img')) {
-                    if (!$images) {
-                        $id = $this->get('imap_msg_part');
-                        $txt .= '<div class="allow_image_link">'.
-                            '<a href="#" class="msg_part_link" data-allow-images="1" '.
-                            'data-message-part="'.$this->html_safe($id).'">'.
-                            $this->trans('Allow Images').'</a></div>';
+                $msgText = $this->get('msg_text');
+                // Everything in the message starting with src="http:// or src="https://
+                $externalResRegexp = '/src="https?:\/\/[^"]*"/i';
+                // Get all external resources
+                preg_match_all($externalResRegexp, $msgText, $externalResources);
+                $externalResources = array_map(function ($resource) {
+                    return preg_replace('/src="(.*)"/i', '$1', $resource);
+                }, $externalResources[0]);
+
+                $externalResourcesText = '<div class="external_notices">';
+                
+                $externalResourcesText .= '</div>';
+                $txt .= $externalResourcesText;
+
+                $replacer = "**000**";
+                $msgText = preg_replace($externalResRegexp, $replacer, $msgText);
+                $msgTextArray = explode($replacer, $msgText);
```

Comment:
Note that you have preg_replace_callback function that allows you to do a replace and execute custom php code while replacing - I think it will make this block easier to follow and also skip the need of custom placeholders like **000**.

Reply:
I really don't understand how preg_replace_callback would help me to get an array whose elements are the values returned from the same function `preg_replace_callback`.

Reply:
Something like:
```
$finalText = preg_replace_callback('/src="(https?:\/\/[^"]*)"/i', function ($matches) use ($allowed) {
    return 'data-src="' . ($allowed ? $matches[1] : '') .'" ' . 'src="" ' . 'data-message-part="'.$this->html_safe($this->get('imap_msg_part')) . '"';
}, $msgText);
```
Also note that src attributes might enclose the links in single quotes.

Reply:
Alright, this is much better. I'm adjusting the regex so that it takes into consideration the links enclosed in single quotes as well.